### ISSUE TITLE
Prepare upcoming text-buffer change

### ISSUE THREAD

**Issue by t9md** [t9md/atom-vim-mode-plus#206]
Discussed in https://github.com/t9md/atom-vim-mode-plus/pull/203

I also tried how the change will be with Atom 1.7.0-beta.

Currently text-buffer not expose Patch class itself, so to use `Patch.compose()`, I need to manually extract it from instance like  `Patch = patch.constructor`.

**Comment by t9md** [comment]
After above change, failing tests is like below.
I think these failure maybe effect of  https://github.com/atom/atom/pull/10878 as @BinaryMuse explained in #203.

![image](https://cloud.githubusercontent.com/assets/155205/13945935/9583784c-f054-11e5-93d1-199a58007d94.png)

**Comment by t9md** [comment]
Sorry I made mess of commit history(I'm still not familiar with git workflow..).
8c4bc87 is commit rebased against master.

**Comment by BinaryMuse** [comment]
Awesome! Yeah, the failing tests you're seeing are a result of the commit I mentioned in the other issue. @kuychaco and I are both out this week for a mini-conference, but I'll talk with her asap about the change and what it entailed.

**Comment by Platzer** [comment]
Hi, I'm not sure if it is releated to this issue, but i recognize that repeating insert is not working with:
```
Atom    : 1.8.0-beta0
Electron: 0.36.8
Chrome  : 47.0.2526.110
Node    : 5.1.1
```
![atominsert](https://cloud.githubusercontent.com/assets/3358663/14520102/f8fdb1d2-0222-11e6-80b3-7ee07717c9ed.gif)

**Comment by t9md** [comment]
Yea, related.
I haven't noticed Atom 1.7 is released.
I first have to care atom 1.7.
I could repro issue in 1.7.
Not sure 1.8-beta.
Will check later.

By the way catching up very fast changing atom with this kind of big package is tough.

**Comment by t9md** [comment]
fixed, released v0.33.0

**Comment by Platzer** [comment]
Thx for the realy quick fix! You build realy great package. I dont wanna push on you or something... Sry for that!
